### PR TITLE
Task/APPS-2047 — Date and Time props

### PR DESCRIPTION
**Pull request by giuliacampana**
Add `WidgetProps` that can be passed to Date, Time, and DateTime Widgets: `ampm`, `format`, `mask`, and `placeholder`.

### CODE REVIEW

File: src/widgets/datetime/DateTimeWidget.js [hunk 1]
```diff
@@ -28,8 +31,11 @@ const DateTimeWidget = (props :WidgetProps) => {
   return (
     <DateTimePicker
         disabled={disabled || readonly}
+        format={format}
         id={id}
+        mask={mask}
         onChange={handleChange}
+        placeholder={placeholder}
         value={value} />
```
Comment (solomontang) [inline src/widgets/datetime/DateTimeWidget.js:39]: missed passing ampm prop

File: src/widgets/datetime/DateTimeWidget.test.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  test('should have ampm in props options', () => {
+    const wrapper = mount(<MuiPickersUtilsProvider utils={LatticeLuxonUtils}><MockForm /></MuiPickersUtilsProvider>);
+    const dateTimeWidget = wrapper.find(DateTimeWidget).at(0);
+    const { options } = dateTimeWidget.props();
+    expect(options.mask).toEqual('__/__/__ __:__');
+  });
+});
```
Comment (solomontang) [inline src/widgets/datetime/DateTimeWidget.test.js:54]: Does not check for ampm or format props from `options` object. Maybe add another separate test for `placeholder` as well.

File: src/widgets/datetime/DateWidget.test.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    const wrapper = mount(<MuiPickersUtilsProvider utils={LatticeLuxonUtils}><MockForm /></MuiPickersUtilsProvider>);
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  test('should have ampm in props options', () => {
+    const wrapper = mount(<MuiPickersUtilsProvider utils={LatticeLuxonUtils}><MockForm /></MuiPickersUtilsProvider>);
+    const dateWidget = wrapper.find(DateWidget).at(0);
+    const { options } = dateWidget.props();
+    expect(options.format).toEqual('yyyy/MM/dd');
+  });
```
Comment (solomontang) [inline src/widgets/datetime/DateWidget.test.js:53]: Just make sure these tests are reflective of what's happening. This test doesn't check for `ampm` even though the description suggests it does.

File: src/widgets/datetime/TimeWidget.test.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+    const wrapper = mount(<MuiPickersUtilsProvider utils={LatticeLuxonUtils}><MockForm /></MuiPickersUtilsProvider>);
+    expect(toJson(wrapper)).toMatchSnapshot();
+  });
+
+  test('should have ampm in props options', () => {
+    const wrapper = mount(<MuiPickersUtilsProvider utils={LatticeLuxonUtils}><MockForm /></MuiPickersUtilsProvider>);
+    const timeWidget = wrapper.find(TimeWidget).at(0);
+    const { options } = timeWidget.props();
+    expect(options.ampm).toEqual(false);
+  });
```
Comment (solomontang) [inline src/widgets/datetime/TimeWidget.test.js:54]: what about the other new props we're passing through?

File: src/widgets/datetime/DateTimeWidget.test.js (hunk 2 above)
Comment (giuliacampana) [reply src/widgets/datetime/DateTimeWidget.test.js:54]: whoops! thank you

File: src/widgets/datetime/TimeWidget.test.js (hunk 4 above)
Comment (giuliacampana) [reply src/widgets/datetime/TimeWidget.test.js:54]: i'll add those!

File: src/widgets/datetime/DateTimeWidget.js (hunk 1 above)
Comment (giuliacampana) [reply src/widgets/datetime/DateTimeWidget.js:39]: making a note here that we need to adjust the `DateTimePicker` in LUK to allow for passing `ampm` prop.

**Pull request merged**